### PR TITLE
[ZEPPELIN-1018] Apply auto "Table of Contents" generator to Zeppelin docs website

### PR DESCRIPTION
**Pull request by AhyoungRyu**
### What is this PR for?
I added auto TOC(Table of Contents) generator for Zeppelin documentation website. TOC can help people looking through whole contents at a glance and finding what they want quickly. 

I just added `<div id="toc"></div>`  to the each documentation header. [`toc`](https://github.com/apache/zeppelin/compare/master...AhyoungRyu:ZEPPELIN-1018?expand=1#diff-85af09fb498a5667ea455391533f945dR3)  recognize `<h2>` & `<h3>` as a title in the docs  and it automatically generate TOC. So I set a rule for this work. (I'll write this rule on `docs/CONTRIBUTING.md` or [docs/howtocontributewebsite](https://zeppelin.apache.org/docs/0.6.0-SNAPSHOT/development/howtocontributewebsite.html)). 

```
# Level-1 Heading  <- Use only for the main title of the page
## Level-2 Heading <- Start with this one
### Level-3 heading <- Only use this one for child of Level-2

toc only recognize Level-2 & Level-3
```

Please see the below attached screenshot image.

### What type of PR is it?
Improvement & Documentation

### Todos
* [x] - Add TOC generator
* [x] - Apply TOC(`<div id="toc"></div>`) to every documentation and reorganize each headers(apply the above rule)
* [x] - Fix some broken code block in several docs
* [x] - Apply TOC to `r.md` (Currently R docs has some duplicated info since [this one](https://github.com/apache/zeppelin/commit/d5e87fb8ba98f08db5b0a4995104ce19f182c678) and [this one](https://github.com/apache/zeppelin/commit/7d6cc7e99154e2d337c11fdf8be1a874ed3e9ada) )
* [x] - Apply TOC to `install.md` after #1010 merged
* [x] - Apply TOC to `interpreterinstallation.md` after #1042 merged

### What is the Jira issue?
[ZEPPELIN-1018](https://issues.apache.org/jira/browse/ZEPPELIN-1018)

### How should this be tested?
1. Apply this patch and build `docs/` with [this guide](https://github.com/apache/zeppelin/tree/master/docs#build-documentation)
2.  Visit some docs page. Then you can see TOC in the header of page. 

### Screenshots (if appropriate)
 - Automatically generated TOC in Spark interpreter docs page
screen shot 2016-06-16 at 9 37 18 pm

### Questions:
* Does the licenses files need update? 
No. Actually I used [jekyll-table-of-contents#copyright](https://github.com/ghiculescu/jekyll-table-of-contents#copyright). But I don't need to add a license for this :)
* Is there breaking changes for older versions? No
* Does this needs documentation? Maybe

### CODE REVIEW

File: docs/development/howtocontributewebsite.md [hunk 1]
```diff
@@ -7,49 +7,52 @@ group: development
...
+Any contributions to Zeppelin (Source code, Documents, Image, Website) means you agree with license all your contributions as Apache2 License.
 
-First of all, you need the website source code. The official location of mirror for Zeppelin is [http://git.apache.org/zeppelin.git](http://git.apache.org/zeppelin.git).
+## Getting the source code
+First of all, you need Zeppelin source code. The official location of Zeppelin is [http://git.apache.org/zeppelin.git](http://git.apache.org/zeppelin.git).
+Documentation website is hosted in 'master' branch under `/docs/` dir.
+
+### git access
 
+First of all, you need the website source code. The official location of mirror for Zeppelin is [http://git.apache.org/zeppelin.git](http://git.apache.org/zeppelin.git).
```
Comment (felixcheung) [inline docs/development/howtocontributewebsite.md]: I think git.apache.org is the actual source code, not a "mirror", but we are told not to have github.com address here.

File: docs/manual/publish.md [hunk 2]
```diff
@@ -45,4 +46,4 @@ For example,
 Finally, you can show off your beautiful visualization results in your website. 
 <center><img src="../assets/themes/zeppelin/img/docs-img/your-website.png" height="90%" width="90%"></center>
 
-> **Note**: To embed the paragraph in a website, Zeppelin needs to be reachable by that website. 
+> **Note**: To embed the paragraph in a website, Apache Zeppelin needs to be reachable by that website. And please use this feature with caution and in a trusted environment only, as Zeppelin entire Webapp could be accessible for whoever visits your website.
```
Comment (felixcheung) [inline docs/manual/publish.md]: hmm, not sure if I understand this correctly.. should this be
"accessible by whoever visits"

File: docs/manual/notebookashomepage.md [hunk 3]
```diff
@@ -19,91 +19,84 @@ limitations under the License.
...
-  ```
-  ####That's it! Open your browser and navigate to zeppelin and see your customized homepage...
+### Restart Zeppelin
+Restart your Zeppelin server
 
+```
+./bin/zeppelin-deamon stop
+./bin/zeppelin-deamon start
+```
+That's it! Open your browser and navigate to Zeppelin and see your customized homepage.
```
Comment (felixcheung) [inline docs/manual/notebookashomepage.md]: Apache Zeppelin?

File: docs/manual/notebookashomepage.md [hunk 4]
```diff
@@ -19,91 +19,84 @@ limitations under the License.
...
 
 <br />
-### Show notebooks list in your custom homepage
-If you want to display the list of notebooks on your custom zeppelin homepage all
+## Show notebooks list in your custom homepage
+If you want to display the list of notebooks on your custom Zeppelin homepage all
```
Comment (felixcheung) [inline docs/manual/notebookashomepage.md]: Apache Zeppelin?

File: docs/interpreter/spark.md [hunk 5]
```diff
@@ -7,8 +7,14 @@ group: manual
...
 
 
-## Spark Interpreter for Apache Zeppelin
-[Apache Spark](http://spark.apache.org) is supported in Zeppelin with
+# Spark Interpreter for Apache Zeppelin
+
+<div id="toc"></div>
+
+## Overview
+[Apache Spark](http://spark.apache.org)  is a fast and general-purpose cluster computing system.
```
Comment (felixcheung) [inline docs/interpreter/spark.md]: not sure if it's visible, there's an extra space after the link

File: docs/interpreter/postgresql.md [hunk 6]
```diff
@@ -44,10 +49,19 @@ Postgresql Interpreter will be deprecated and merged into JDBC Interpreter. You
...
+## Overview
+
+[<img align="right" src="http://img.youtube.com/vi/wqXXQhJ5Uk8/0.jpg" alt="zeppelin-view" hspace="10" width="250"></img>](https://www.youtube.com/watch?v=wqXXQhJ5Uk8)
 
-----
+This interpreter seamlessly supports the following SQL data processing engines:
 
-## PostgreSQL, HAWQ  Interpreter for Apache Zeppelin
+* [PostgreSQL](http://www.postgresql.org/) - OSS, Object-relational database management system (ORDBMS)
+* [Apache HAWQ](http://pivotal.io/big-data/pivotal-hawq) - Powerful [Open Source](https://wiki.apache.org/incubator/HAWQProposal) SQL-On-Hadoop engine.
```
Comment (felixcheung) [inline docs/interpreter/postgresql.md]: probably need to update this to `Apache HAWQ (incubating)`
and http://hawq.incubator.apache.org/

File: docs/interpreter/postgresql.md [hunk 7]
```diff
@@ -6,7 +6,12 @@ group: manual
 ---
 {% include JB/setup %}
 
+# PostgreSQL, HAWQ  Interpreter for Apache Zeppelin
```
Comment (felixcheung) [inline docs/interpreter/postgresql.md]: probably need to update HAWQ to Apache HAWQ (incubating)

File: docs/interpreter/livy.md [hunk 8]
```diff
@@ -6,8 +6,12 @@ group: manual
...
 {% include JB/setup %}
 
-## Livy Interpreter for Apache Zeppelin
-Livy is an open source REST interface for interacting with Spark from anywhere. It supports executing snippets of code or programs in a Spark context that runs locally or in YARN.
+# Livy Interpreter for Apache Zeppelin
+
+<div id="toc"></div>
+
+## Overview
+[Livy](http://livy.io/) is an open source REST interface for interacting with Spark from anywhere. It supports executing snippets of code or programs in a Spark context that runs locally or in YARN.
```
Comment (felixcheung) [inline docs/interpreter/livy.md]: nice. looks like they have a logo on 
http://livy.io/img/livy-logo.png

File: docs/development/howtocontributewebsite.md (hunk 1 above)
Comment (AhyoungRyu) [inline docs/development/howtocontributewebsite.md]: @felixcheung I'm not sure I understood incorrectly, As you can see in [git.apache.org](http://git.apache.org/), in the first sentence, they said 
>This is a collection of read-only Git mirrors of Apache codebases.

Am I missing sth ?

File: docs/interpreter/livy.md (hunk 8 above)
Comment (AhyoungRyu) [inline docs/interpreter/livy.md]: @felixcheung Yeah right. Maybe next time we can update the available interpreters image in `index.md` :)

**Comment by AhyoungRyu** [comment]
@felixcheung Really appreciate your feedback :) I just addressed them.

**Comment by AhyoungRyu** [comment]
TODO list is done for now. This PR has big changes in overall documentation, but most of them is for reorganization the title & header in each docs for more well-organized TOC. 
So ready for review :)

**Comment by AhyoungRyu** [comment]
But CI failure is irrelevant with this change i think. There are 2 errors like below.

#### 1.

Running org.apache.zeppelin.python.PythonInterpreterTest
SLF4J: Class path contains multiple SLF4J bindings.
SLF4J: Found binding in [jar:file:/home/travis/build/apache/zeppelin/zeppelin-interpreter/target/zeppelin-interpreter-0.6.0-SNAPSHOT.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: Found binding in [jar:file:/home/travis/build/apache/zeppelin/zeppelin-interpreter/target/lib/slf4j-log4j12-1.7.10.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: Found binding in [jar:file:/home/travis/.m2/repository/org/slf4j/slf4j-log4j12/1.7.10/slf4j-log4j12-1.7.10.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: See http://www.slf4j.org/codes.html#multiple_bindings for an explanation.
SLF4J: Actual binding is of type [org.slf4j.impl.Log4jLoggerFactory]
log4j:WARN No appenders could be found for logger (org.apache.zeppelin.interpreter.Interpreter).
log4j:WARN Please initialize the log4j system properly.
log4j:WARN See http://logging.apache.org/log4j/1.2/faq.html#noconfig for more info.
Tests run: 5, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 0.434 sec <<< FAILURE! - in org.apache.zeppelin.python.PythonInterpreterTest
testClose(org.apache.zeppelin.python.PythonInterpreterTest)  Time elapsed: 0.317 sec  <<< FAILURE!
java.lang.AssertionError: null
    at org.junit.Assert.fail(Assert.java:86)
    at org.junit.Assert.assertTrue(Assert.java:41)
    at org.junit.Assert.assertFalse(Assert.java:64)
    at org.junit.Assert.assertFalse(Assert.java:74)
    at org.apache.zeppelin.python.PythonInterpreterTest.testClose(PythonInterpreterTest.java:170)

Results :

Failed tests:
  PythonInterpreterTest.testClose:170 null

Tests run: 5, Failures: 1, Errors: 0, Skipped: 0

[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary:
[INFO]
[INFO] Zeppelin ........................................... SUCCESS [ 10.254 s]
[INFO] Zeppelin: Interpreter .............................. SUCCESS [01:55 min]
[INFO] Zeppelin: Zengine .................................. SUCCESS [ 24.280 s]
[INFO] Zeppelin: Display system apis ...................... SUCCESS [  2.810 s]
[INFO] Zeppelin: Spark dependencies ....................... SUCCESS [ 42.031 s]
[INFO] Zeppelin: Spark .................................... SUCCESS [01:47 min]
[INFO] Zeppelin: Markdown interpreter ..................... SUCCESS [  0.634 s]
[INFO] Zeppelin: Angular interpreter ...................... SUCCESS [  0.285 s]
[INFO] Zeppelin: Shell interpreter ........................ SUCCESS [  0.391 s]
[INFO] Zeppelin: Livy interpreter ......................... SUCCESS [  0.612 s]
[INFO] Zeppelin: HBase interpreter ........................ SUCCESS [  9.700 s]
[INFO] Zeppelin: PostgreSQL interpreter ................... SUCCESS [  1.399 s]
[INFO] Zeppelin: JDBC interpreter ......................... SUCCESS [  2.255 s]
[INFO] Zeppelin: File System Interpreters ................. SUCCESS [  1.872 s]
[INFO] Zeppelin: Flink .................................... SUCCESS [ 19.149 s]
[INFO] Zeppelin: Apache Ignite interpreter ................ SUCCESS [ 22.431 s]
[INFO] Zeppelin: Kylin interpreter ........................ SUCCESS [  0.937 s]
[INFO] Zeppelin: Python interpreter ....................... FAILURE [  1.325 s]
[INFO] Zeppelin: Lens interpreter ......................... SKIPPED
[INFO] Zeppelin: Apache Cassandra interpreter ............. SKIPPED
[INFO] Zeppelin: Elasticsearch interpreter ................ SKIPPED
[INFO] Zeppelin: Alluxio interpreter ...................... SKIPPED
[INFO] Zeppelin: web Application .......................... SKIPPED
[INFO] Zeppelin: Server ................................... SKIPPED
[INFO] Zeppelin: Packaging distribution ................... SKIPPED
[INFO] Zeppelin: R Interpreter ............................ SKIPPED
[INFO] Zeppelin: Scalding interpreter ..................... SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 06:03 min
[INFO] Finished at: 2016-06-21T07:17:53+00:00
[INFO] Final Memory: 59M/357M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.17:test (default-test) on project zeppelin-python: There are test failures.
[ERROR]
[ERROR] Please refer to /home/travis/build/apache/zeppelin/python/target/surefire-reports for the individual test results.
[ERROR] -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
[ERROR]
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <goals> -rf :zeppelin-python

#### 2. Selenium Test error
Results :

Failed tests:
  ParagraphActionsIT.testTitleButton:350 After Show Title : The title field contains
Expected: "Untitled"
     but: was ""
  ParagraphActionsIT.testTitleButton:360 After Hide Title : The title field contains
Expected: ""
     but: was "Untitled"
  ParagraphActionsIT.testRemoveButton:161 After Remove : Number of paragraphs are
Expected: <2>
     but: was <1>

Tests run: 16, Failures: 3, Errors: 0, Skipped: 0

[INFO]
[INFO] --- scalatest-maven-plugin:1.0:test (test) @ zeppelin-server ---
[36mDiscovery starting.[0m
Jun 21, 2016 7:18:12 AM org.openqa.selenium.safari.SafariDriverServer start
INFO: Server started on port 10154
Jun 21, 2016 7:18:12 AM org.openqa.selenium.safari.SafariDriverCommandExecutor stop
INFO: Shutting down
Jun 21, 2016 7:18:12 AM org.openqa.selenium.safari.SafariDriverCommandExecutor stop
INFO: Stopping server
Jun 21, 2016 7:18:12 AM org.openqa.selenium.safari.SafariDriverServer stop
INFO: Stopping server
Jun 21, 2016 7:18:12 AM org.openqa.selenium.safari.SafariDriverCommandExecutor stop
INFO: Shutdown complete
[36mDiscovery completed in 3 seconds, 135 milliseconds.[0m
[36mRun starting. Expected test count is: 1[0m
Zeppelin is already running
[32mWelcomePageSuite:[0m
[32m- Welcome sign is correct[0m
Zeppelin stop [60G[[1;32m  OK  [0;39m]
[36mRun completed in 12 seconds, 44 milliseconds.[0m
[36mTotal number of tests run: 1[0m
[36mSuites: completed 3, aborted 0[0m
[36mTests: succeeded 1, failed 0, canceled 0, ignored 0, pending 0[0m
[32mAll tests passed.[0m
[INFO]
[INFO] --- maven-antrun-plugin:1.7:run (stop-zeppelin) @ zeppelin-server ---
[INFO] Executing tasks

main:
     [exec] Zeppelin is not running
[INFO] Executed tasks
[INFO]
[INFO] --- apache-rat-plugin:0.11:check (verify.rat) @ zeppelin-server ---
[INFO] 51 implicit excludes (use -debug for more details).
[INFO] Exclude: **/.idea/
[INFO] Exclude: **/*.iml
[INFO] Exclude: .git/
[INFO] Exclude: .gitignore
[INFO] Exclude: **/.settings/*
[INFO] Exclude: **/.classpath
[INFO] Exclude: **/.project
[INFO] Exclude: **/target/**
[INFO] Exclude: **/derby.log
[INFO] Exclude: **/metastore_db/
[INFO] Exclude: **/README.md
[INFO] Exclude: src/test/java/com/webautomation/*
[INFO] 60 resources included (use -debug for more details)
[INFO] Rat check: Summary of files. Unapproved: 0 unknown: 0 generated: 0 approved: 60 licence.
[INFO]
[INFO] --- maven-failsafe-plugin:2.16:verify (default) @ zeppelin-server ---
[INFO] Failsafe report directory: /home/travis/build/apache/zeppelin/zeppelin-server/target/failsafe-reports
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary:
[INFO]
[INFO] Zeppelin: Interpreter .............................. SUCCESS [ 21.471 s]
[INFO] Zeppelin: Zengine .................................. SUCCESS [  5.803 s]
[INFO] Zeppelin: Display system apis ...................... SUCCESS [  3.944 s]
[INFO] Zeppelin: Spark dependencies ....................... SUCCESS [ 37.600 s]
[INFO] Zeppelin: Spark .................................... SUCCESS [  4.322 s]
[INFO] Zeppelin: Server ................................... FAILURE [07:21 min]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 08:35 min
[INFO] Finished at: 2016-06-21T07:18:25+00:00
[INFO] Final Memory: 47M/337M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-failsafe-plugin:2.16:verify (default) on project zeppelin-server: There are test failures.
[ERROR]
[ERROR] Please refer to /home/travis/build/apache/zeppelin/zeppelin-server/target/failsafe-reports for the individual test results.
[ERROR] -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
[ERROR]
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <goals> -rf :zeppelin-server

Are they known errors?

File: docs/assets/themes/zeppelin/js/toc.js [hunk 9]
```diff
@@ -0,0 +1,98 @@
+// https://github.com/ghiculescu/jekyll-table-of-contents
```
Comment (Leemoonsoo) [inline docs/assets/themes/zeppelin/js/toc.js]: Could you add license entry of toc.js into `LICENSE` file and make a copy of license under `licenses` directory, while this file will be included in our source distribution?

Comment (AhyoungRyu) [inline docs/assets/themes/zeppelin/js/toc.js]: @Leemoonsoo Yeah I added the license info as you said. Please see [f704b5e](https://github.com/apache/zeppelin/pull/1031/commits/f704b5e6eacd04ac7e94e41736e42dd6b8af46ba).

**Comment by minahlee** [comment]
It looks great to me. Thanks for improving document

**Comment by AhyoungRyu** [comment]
@minahlee My pleasure, thanks :)

**Pull request closed without merging**